src/main/java/net/mcreator/util/BlocklyHelper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2022, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.util;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class BlocklyHelper {
+	Document doc;

```

Comment:
I think you can change the new class to record as you don't re-assign this field anywhere.

Reply:
I don't think that's required. This class only provides utility methods, and doesn't expose the `doc` field

Reply:
It could be private final